addons/web/static/tests/core/code_editor.test.js

```diff
@@ -0,0 +1,276 @@
+        setTheme: () => {},
+        resize: () => {},
+        destroy: () => {},
+        setSession: () => {},
+        getSession() {
+            return this.session;
+        },
+        on: () => {},
+    };
+}
+
+test("Can be rendered", async () => {
+    class Parent extends Component {
+        static components = { CodeEditor };
+        static template = xml`<CodeEditor mode="'xml'" />`;
+        static props = ["*"];
+    }
+    await mountWithCleanup(Parent);
+    expect(".ace_editor").toHaveCount(1);
+});
+
+test("CodeEditor shouldn't accepts markup values", async () => {
+    onError((ev) => {
+        if (ev instanceof PromiseRejectionEvent) {
+            ev.preventDefault();
+        }
+    });
+
+    const _console = window.console;
+    window.console = Object.assign(Object.create(_console), {
+        warn(msg) {
+            expect.step(msg);
+        },
+    });
+    after(() => {
+        window.console = _console;
+    });
+
+    class Parent extends Component {
+        static components = { CodeEditor };
+        static template = xml`<CodeEditor value="props.value"/>`;
+        static props = ["*"];
+    }
+    class GrandParent extends Component {
+        static components = { Parent };
+        static template = xml`<Parent value="state.value"/>`;
+        static props = ["*"];
+        setup() {
+            this.state = useState({ value: `<div>Some Text</div>` });
+        }
+    }
+
+    const codeEditor = await mountWithCleanup(GrandParent);
+    const textMarkup = markup`<div>Some Text</div>`;
+
+    codeEditor.state.value = textMarkup;
+    await animationFrame();
+    expect(["Invalid props for component 'CodeEditor': 'value' is not valid"]).toVerifyErrors();
+    expect(["[Owl] Unhandled error. Destroying the root component"]).toVerifySteps();
+});
+
+test("onChange props called when code is edited", async () => {
+    class Parent extends Component {
+        static components = { CodeEditor };
+        static template = xml`<CodeEditor onChange.bind="onChange" />`;
+        static props = ["*"];
+        onChange(value) {
+            expect.step(value);
+        }
+    }
+
+    await mountWithCleanup(Parent);
+    await editAce("Some Text");
+    expect([
+        "S",
+        "So",
+        "Som",
+        "Some",
+        "Some ",
+        "Some T",
+        "Some Te",
+        "Some Tex",
+        "Some Text",
+    ]).toVerifySteps();
+});
+
+test("onChange props not called when value props is updated", async () => {
+    class Parent extends Component {
+        static components = { CodeEditor };
+        static template = xml`<CodeEditor value="state.value" onChange.bind="onChange" />`;
+        static props = ["*"];
+        state = useState({ value: "initial value" });
+        onChange(value) {
+            expect.step(value || "__emptystring__");
+        }
+    }
+
+    const parent = await mountWithCleanup(Parent);
+    expect(".ace_line").toHaveText("initial value");
```

Comment:
we had an extra nextTick before, any idea why?

Reply:
yes, in mountWithCleanup, by default we wait for or mainComponentContainer to be mounted, which yields
the other animationframe we, in any case, want to wait for